### PR TITLE
move unused functions from Cardano.Crypto.Limits.hs to Test.Cardano.Crypto.Limits.hs

### PR DESCRIPTION
**Pull request by DeepakKapiswe**
Drop `Cardano.Crypto.Limits.hs`
Addresses https://github.com/input-output-hk/cardano-base/pull/25

### CODE REVIEW

**Review by ruhatch — APPROVED** [review]
LGTM!

File: crypto/test/Test/Cardano/Crypto/Limits.hs [hunk 1]
```diff
@@ -30,6 +36,38 @@ import Test.Cardano.Crypto.Gen
 tests :: IO Bool
 tests = checkParallel $$discover
 
+---------------------------------------------------------------------------
+-- Limit
+---------------------------------------------------------------------------
+-- | A limit on the length of something (in bytes).
```
Comment (dnadales) [inline crypto/test/Test/Cardano/Crypto/Limits.hs:42]: Maybe "a limit on the length of a data value"?

File: crypto/test/Test/Cardano/Crypto/Limits.hs [hunk 2]
```diff
@@ -30,6 +36,38 @@ import Test.Cardano.Crypto.Gen
...
+-- Helper Functions
+--------------------------------------------------------------------------------
+
+mlAbstractHash
+  :: forall algo a . HashAlgorithm algo => Limit (AbstractHash algo a)
+mlAbstractHash =
+  fromIntegral (hashDigestSize (panic "AbstractHash limit" :: algo) + 4)
+
+mlVerificationKey :: Limit VerificationKey
+mlVerificationKey = 66
```
Comment (dnadales) [inline crypto/test/Test/Cardano/Crypto/Limits.hs:64]: We might want to explain why did we chose these constants.

File: crypto/test/Test/Cardano/Crypto/Limits.hs [hunk 3]
```diff
@@ -30,6 +36,38 @@ import Test.Cardano.Crypto.Gen
...
+  fmap _ (Limit x) = Limit x
+
+--------------------------------------------------------------------------------
+-- Helper Functions
+--------------------------------------------------------------------------------
+
+mlAbstractHash
+  :: forall algo a . HashAlgorithm algo => Limit (AbstractHash algo a)
+mlAbstractHash =
+  fromIntegral (hashDigestSize (panic "AbstractHash limit" :: algo) + 4)
```
Comment (dnadales) [inline crypto/test/Test/Cardano/Crypto/Limits.hs:61]: Might be worth explaining why did we add a `panic` here. And why not just using `undefined`?

File: crypto/test/Test/Cardano/Crypto/Limits.hs [hunk 4]
```diff
@@ -30,6 +36,38 @@ import Test.Cardano.Crypto.Gen
 tests :: IO Bool
 tests = checkParallel $$discover
 
+---------------------------------------------------------------------------
+-- Limit
+---------------------------------------------------------------------------
+-- | A limit on the length of something (in bytes).
+--   TODO should check for overflow in the Num instance.
```
Comment (dnadales) [inline crypto/test/Test/Cardano/Crypto/Limits.hs:43]: Hmmm, I'm not sure about this comment. I'd rather say that we support maximum sizes equal to `maxBound :: Word32`. Maybe we would like to clarify that `maxBound` bytes is about 4GB.

**Review by dnadales — COMMENTED** [review]
Looks OK to me, but I think we would want to add more comments, specially around the magic constants and `panic` usages.

Since I don't have ownership on this codebase I'd feel more comfortable if @erikd or @ruhatch could take a quick look.

**Review by ruhatch — COMMENTED** [review]
bors r+

**Pull request merged**